assets/src/edit-story/components/checklist/checks/videoOptimization.js

```diff
@@ -0,0 +1,199 @@
+  }, [unoptimizedVideos, optimizeVideo]);
+
+  const handleUpdateVideos = useCallback(async () => {
+    unoptimizedVideos.forEach(({ resource }) => {
+      if (!resource.isTranscoding) {
+        optimizingResourcesRef.current[resource.id] = 'UPLOADING';
+      }
+    });
+    await sequencedVideoOptimization();
+  }, [sequencedVideoOptimization, unoptimizedVideos]);
+
+  const handleClickThumbnail = useCallback(
+    (element) => async () => {
+      setHighlights({
+        pageId: element.pageId,
+        elementId: element.id,
+      });
+      if (
+        !element.resource?.isTranscoding &&
+        !['UPLOADING'].includes(
+          optimizingResourcesRef.current[element.resource.id]
+        )
+      ) {
+        optimizingResourcesRef.current[element.resource.id] = 'UPLOADING';
+        await optimizeVideo({ resource: element.resource }).finally(() => {
+          optimizingResourcesRef.current[element.resource.id] = 'UPLOADED';
+        });
+      }
+    },
+    [optimizeVideo, setHighlights]
+  );
+
+  const { footer, title } = PRIORITY_COPY.videoNotOptimized;
+
+  const isTranscoding = useMemo(
+    () =>
+      Object.values(optimizingResourcesRef.current).includes('UPLOADING') ||
+      unoptimizedVideos.some((video) => video.resource?.isTranscoding),
+    [unoptimizedVideos]
+  );
+
+  const isRendered = unoptimizedVideos.length > 0;
+  useRegisterCheck('VideoOptimization', isRendered);
+
+  return (
+    isRendered && (
+      <ChecklistCard
+        cta={
+          <OptimizeButton
+            type={BUTTON_TYPES.SECONDARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={handleUpdateVideos}
+            disabled={isTranscoding}
+          >
+            {isTranscoding
+              ? _n(
+                  'Optimizing video',
+                  'Optimizing videos',
+                  unoptimizedVideos.length,
+                  'web-stories'
+                )
+              : _n(
+                  'Optimize video',
+                  'Optimize all videos',
+                  unoptimizedVideos.length,
+                  'web-stories'
+                )}
+          </OptimizeButton>
+        }
+        title={title}
+        cardType={
+          unoptimizedVideos.length > 1
+            ? CARD_TYPE.MULTIPLE_ISSUE
+            : CARD_TYPE.SINGLE_ISSUE
+        }
+        footer={<DefaultFooterText>{footer}</DefaultFooterText>}
+        thumbnailCount={unoptimizedVideos.length}
+        thumbnail={unoptimizedVideos.map((element) => (
+          <Thumbnail
+            key={element.id}
+            onClick={handleClickThumbnail(element)}
+            isLoading={element.resource?.isTranscoding}
+            type={THUMBNAIL_TYPES.VIDEO}
+            displayBackground={<LayerThumbnail page={element} />}
+            aria-label={__('Go to offending video', 'web-stories')}
+          >
+            {!['UPLOADING'].includes(
+              optimizingResourcesRef.current[element.resource?.id]
+            ) && (
```

Comment:
You shouldn't need this check here because the tooltip can only ever render if `isLoading` and `isError` are both `false`.